README.md

```diff
@@ -329,37 +348,79 @@ There MUST be at least one path segment as a namespace. For example, `http/PUT`
 
 The only reserved ability MUST be the un-namespaced [`"*"` or "superuser"](#41-superuser), which MUST be allowed on any resource.
 
-#### Examples
+### 3.2.5.3 `nb` Non-Normative Fields
+
+Capabilities MAY define additional optional or required fields specific to their use case in the `nb` ([nota bene](https://en.wikipedia.org/wiki/Nota_bene)) field. This field is OPTIONAL in the general case, but MAY be REQUIRED by particular capability types that require this information to validate. The `nb` field MAY contain additional caveats or other important information related to specifying the capability, and MAY function as an "escape hatch" for when a use case is not fully captured by the `with` and `can` fields.
+
+If a validator is not able to interpret the `nb` field, it MUST reject the capability.
```

Comment:
As side note I think I like how [ZCAP-LD describes these as "caveats"](https://w3c-ccg.github.io/zcap-spec/#caveats)

Crucially they do note that "caveats" are inherited from parents. Here it is unclear if caveats get inherited / merged through delegation chains or whether they need to be repeated and whether omition would constitute escalation.

Reply:
In other words I think it would be good to clarify whether `nb` gets inherited and if there are some predefined merge semantics for cases where claimed capability introduces new field and does not include prior fields.

Reply:
From the example below I'm not certain what would following constitute:

```ts
"att": [
  {
    "with": "mailto:username@example.com",
    "can": "msg/READ",
    "nb": {
      "templates": ["newsletter", "marketing"]
    }
  }
],
"prf": [
   // ...
   "att": [
      {
        "with": "mailto:username@example.com",
        "can": "msg/READ",
        "nb": {
          "max_count": 5,
        }
      }
   ]
]
```

Reply:
And I suppose things can get even trickier when you may have two proofs one delegating access to `"templates": ["newsletter"]` and other to `"templates": ["marketing"]`.

Reply:
> In other words I think it would be good to clarify whether nb gets inherited and if there are some predefined merge semantics for cases where claimed capability introduces new field and does not include prior fields.

Okay yeah, totally! In UCAN, every delegation is technically a caveat on the parent, but `nb` doesn't have to only include caveats. Capabilities taken as a single unit have caveat semantics, and are checked as the UCAN is validated (you don't need to forward them into the top level, they need to be spelled out at each delegation). If you embedded a non-UCAN proof (e.g. shared secret for that capability, or a SIWE token), that's not related to caveats per se. It's "just" related information.

Let me think on this a bit. It may be nice to have those separated for clarity.

Reply:
> And I suppose things can get even trickier when you may have two proofs one delegating access to "templates": ["newsletter"] and other to "templates": ["marketing"]

Yup totally. The validation is constructive, so you can merge those capabilities for a downstream proof. Extra caveats on one template would not automatically apply to the other (since they could use them separately).

Reply:
> If you embedded a non-UCAN proof (e.g. shared secret for that capability, or a SIWE token), that's not related to caveats per se. It's "just" related information.

I think it depends on how you define "caveats". I'm under impression that you associate them with restrictions, however according to this following dictionary definitions, I'd argue that any additional details  inculding non-UCAN proofs would are still caveats:

> - an explanation to prevent misinterpretation
> - a [modifying](https://www.merriam-webster.com/dictionary/modify) or cautionary detail to be considered when evaluating, interpreting, or doing something

Reply:
"Caveat" seems to have a narrow technical meaning in a lot of capability systems. We can choose to break with this convention, but just for reference:

Macaroons:
> With Macaroons, a service can attenuate the token before sending it to the next service, by adding a caveat, a condition over the current request (expiration date, limiting to read operations, restricting file paths to a prefix…)

zcap-ld:
>  "Caveats" may be attached to capability documents which may be used to restrict the scope of their use

Reply:
[Updated the PR](https://github.com/ucan-wg/spec/pull/55/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R355) to be even clearer that contained caveats MUST be checked, and thus attenuate downstream caps.

I think that we're using the same definition for "caveats", but that the `nb` field is currently broader than "just" caveats. Should we split these fields into "must checkable" and not? I'm unsure that there's a clear need to do this, but maybe!

Reply:
Hmm I also wonder if things like templates should tap into the URI syntax? Thoughts? I've adjusted your example for this below:

![](https://upload.wikimedia.org/wikipedia/commons/thumb/d/d6/URI_syntax_diagram.svg/2136px-URI_syntax_diagram.svg.png)

```js
"att": [
  {
    "with": "mailto:username@example.com?template=newsletter&template=marketing",
    "can": "msg/READ"
  }
],
```

Reply:
It feels like there's a distinction here between the resource (e.g. email, templates, etc), and information like `max_count` 🤔